### ISSUE TITLE
On Hold/Pending Queue

### ISSUE THREAD

**Issue by horusofoz** [HabitRPG/habitica#243]
I think it would be beneficial to add a pending flag for Todos which cannot be actioned due to external factors (time, resources not available, waiting on someone else, etc).

The list of tasks Pending would be accessed via a view in the same style as Completed Todos currently are. While a task is Pending, it would not affect health nor accrue value.

This "Pending flag/queue" would  allow users to flag a Todo as Pending under Edit. Once flagged a mandatory text field would become visible to add a reason why the action is pending and (not mandatory) select a Start Date for task to be moved to active Todos view.

**Comment by horusofoz** [comment]
I haven't moved this to Trello at this stage as want to see if there is anyone else interested before doing so.

**Comment by ghost** [comment]
Care to elaborate on the overall idea? Isn't this the point of todos?

**Comment by wc8** [comment]
I think the idea is to mark todos as on-hold, freezing their value and (if disease is implemented) potential for harm.

**Comment by horusofoz** [comment]
@wc8 That is it. To be able to mark todos/dailys as on hold when they cannot be performed for reason X.

**Comment by horusofoz** [comment]
On hold would be the status for tasks pending their due date or awaiting external action, etc.

**Comment by horusofoz** [comment]
I know this isn't an exciting feature but I think it is essential for work groups and/or incorporating HabitRPG into workflows. I've raised as [Trello 135](https://trello.com/c/exaF7GiC) to further gauge interest.

**Comment by wc8** [comment]
@horusofoz as I was looking at my todo list, I saw a number of todos that are deep red, but cannot be done because I am out of state. It would be nice to be able to lock them into a grey, inactive/on hold mode until I return. 

Thinking about this, your suggestion covers more than a simple start date (external factors (time, resources not available, waiting on someone else, etc) but seems to involve applying a start date to all of them. Is this basically just an enhancement of the start date request (currently a sub-request listed under [Urgency: Dates & Deadlines](https://trello.com/c/i6kXvNY2)) or would there be cases in which on-hold would apply without a start date? 

Could you elaborate on this some more?

**Comment by horusofoz** [comment]
This applies to jobs also without a start date. e.g. Fix Rahim's PC has no Start Date as I don't know when he will come by. ON Hold though so I have a note of it and can check my on hold work whenever running short of active tasks.

**Comment by horusofoz** [comment]
Also the Start Date field is NOT mandatory. A reason for the task being on hold is.

**Comment by wc8** [comment]
@horusofoz I'd like to propose a new synopsis for the Trello card. 

Description: 

> A setting to mark a todo as inactive.
> Inactive tasks would not accrue in value, cause harm health, or be vulnerable to disease. _They can be completed at anytime. (HP gain would be as if the task had just been entered.) They can be activated at anytime._
> 
> User _must_ provide a reason/obstacle (text field): 
> - waiting on [potentially a party member], 
> - resources not available
> - time
> - needs planning
> - etc.
> 
> User _may_ specify a Start Date at which the task becomes active.
> 
> This would ideally be implemented with status views for todos and daily tasks (_All, Active, On Hold, Completed_)
> On hold/inactive tasks would be shaded grey. 

I included your views suggestion and added a couple of sentences (see _italics_) about completing and activating. I removed the suggestion that it's not an exciting feature. 

If any of this seems useful, feel free to incorporate it.

(Part of the reason On Hold might not have many votes, though, is that todos currently pose no harm, they become more valuable with time.)

**Comment by horusofoz** [comment]
@wc8 Go for it. Better worded than I had it. Thanks :)
